h5py/tests/test_dtype.py

```diff
@@ -109,44 +115,45 @@ def a(items):
     def test_vlen_enum(self):
         fname = self.mktemp()
         arr1 = [[1], [1, 2]]
-        dt1 = h5py.vlen_dtype(h5py.enum_dtype(dict(foo=1, bar=2), 'i'))
+        dt1 = vlen_dtype(enum_dtype(dict(foo=1, bar=2), 'i'))
 
-        with h5py.File(fname, 'w') as f:
+        with File(fname, 'w') as f:
             df1 = f.create_dataset('test', (len(arr1),), dtype=dt1)
             df1[:] = np.array(arr1)
 
-        with h5py.File(fname, 'r') as f:
+        with File(fname, 'r') as f:
             df2 = f['test']
             dt2 = df2.dtype
             arr2 = [e.tolist() for e in df2[:]]
 
         self.assertEqual(arr1, arr2)
-        self.assertEqual(h5py.check_enum_dtype(h5py.check_vlen_dtype(dt1)),
-                         h5py.check_enum_dtype(h5py.check_vlen_dtype(dt2)))
+        self.assertEqual(check_enum_dtype(check_vlen_dtype(dt1)),
+                         check_enum_dtype(check_vlen_dtype(dt2)))
 
 
 class TestEmptyVlen(TestCase):
     def test_write_empty_vlen(self):
-        fname = self.mktemp()
-        with h5py.File(fname, 'w') as f:
-            d = np.core.records.fromarrays([[], []], names='a,b', formats='|V16,O')
-            dset = f.create_dataset('test', data=d, dtype=[('a', '|V16'), ('b', h5py.special_dtype(vlen=np.float_))])
-            self.assertEqual(dset.size, 0)
+        d = np.core.records.fromarrays([[], []], names='a,b', formats='|V16,O')
+        dset = self.f.create_dataset(
+            'test', data=d, dtype=[
+                ('a', '|V16'), ('b', vlen_dtype(np.float_))
+            ]
+        )
+        self.assertEqual(dset.size, 0)
 
 
 class TestExplicitCast(TestCase):
     def test_f2_casting(self):
-        fname = self.mktemp()
 
         np.random.seed(1)
         A = np.random.rand(1500, 20)
 
         # Save to HDF5 file
-        with h5py.File(fname, "w") as Fid:
-            Fid.create_dataset("Data", data=A, dtype='f2')
+        self.f.create_dataset("Data", data=A, dtype='f2')
+        fname = self.f.filename
 
-        with h5py.File(fname, "r") as Fid:
-            B = Fid["Data"][:]
+        with File(fname, "r") as f:
```

Comment:
Don't we need to close the writeable file handle before opening it again for reading?

Reply:
Yep, I'll fix that.